src/core/ddsi/include/dds/ddsi/q_entity.h

```diff
@@ -641,6 +654,10 @@ void rebuild_or_clear_writer_addrsets(struct q_globals *gv, int rebuild);
 
 void local_reader_ary_setfastpath_ok (struct local_reader_ary *x, bool fastpath_ok);
 
+void connect_writer_with_proxy_reader_secure(struct writer *wr, struct proxy_reader *prd, nn_mtime_t tnow);
```

Comment:
In the case that the endpoint requires payload encoding or submessage encoding then communication can only start when the crypto tokens from the remote entity are available. In that case the matching is delayed and when the associated crypto tokens become available this function is used to complete the matching.

Reply:
When checking for a match it could be that although the match is allowed the crypto tokens which are used for the communication between those endpoints are not yet received. In that case the matching is delayed until the crypto tokens are received and then this function is called match those endpoints.